### PR TITLE
Update hcsshim to latest - v0.8.9

### LINKED ISSUE #87420: hcsshim has recursive dependency on kubernetes
Please see:
- https://github.com/microsoft/hcsshim/blob/master/go.mod#L42
- https://github.com/kubernetes/kubernetes/blob/master/go.mod#L190

Leads to weird stuff:
- https://github.com/microsoft/hcsshim/pull/700
- https://github.com/microsoft/hcsshim/issues/642

Please break the loop. We can't have it both ways!

/sig windows
/assign @PatrickLang

### PR DESCRIPTION
**Pull request by dims**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Use latest release instead of a random SHA. This new version of hcsshim also breaks the recursive dependency between the two projects

**Which issue(s) this PR fixes**:

Fixes #87420

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by liggitt** [comment]
mechanics lgtm

**Comment by liggitt** [comment]
filed https://github.com/kubernetes/kubernetes/issues/89712 for flake
/retest

**Comment by cheftako** [comment]
Change so far seems fine to me.

**Comment by dims** [comment]
cc @dcantah

File: vendor/github.com/Microsoft/hcsshim/internal/schema2/properties.go [hunk 1]
```diff
@@ -44,4 +47,8 @@ type Properties struct {
 	SharedMemoryRegionInfo []SharedMemoryRegionInfo `json:"SharedMemoryRegionInfo,omitempty"`
 
 	GuestConnectionInfo *GuestConnectionInfo `json:"GuestConnectionInfo,omitempty"`
+
+	// Metrics is not part of the API for HCS but this is used for LCOW v2 to
+	// return the full cgroup metrics from the guest.
+	Metrics *v1.Metrics `json:"LCOWMetrics,omitempty"`
```
Comment (dims) [inline vendor/github.com/Microsoft/hcsshim/internal/schema2/properties.go:53]: @dcantah this unnecessarily drags in `github.com/containerd/cgroups/stats/v1` can we somehow avoid this?

Comment (dcantah) [reply vendor/github.com/Microsoft/hcsshim/internal/schema2/properties.go:53]: Definitely doable, need to look into it though.

**Review by elweb9858 — APPROVED** [review]

File: vendor/github.com/Microsoft/hcsshim/internal/schema2/properties.go (hunk 1 above)
Comment (dims) [reply vendor/github.com/Microsoft/hcsshim/internal/schema2/properties.go:53]: Ack @dcantah

**Comment by dims** [comment]
@benmoss @marosset @ddebroy Can one of you please approve this change from sig-windows side?

**Comment by dims** [comment]
@liggitt this is ready for root approval. please take a look.

-- Dims

**Comment by elweb9858** [comment]
> @liggitt this is ready for root approval. please take a look.
> 
> -- Dims

bump @liggitt

**Pull request merged**